### DISCUSSION TITLE
trame v3 is out...

### DISCUSSION THREAD

**Discussion by jourdain** [Kitware/trame#295] [Announcements]
And I most likely broke your project if you did not bound the version of trame in your dependency.

But don't worry, upgrading your project is going to be easy.

1. The new version of trame is removing its implicit dependencies to external widgets and therefore require your application to list the widgets that you are leveraging (i.e.: vtk, vuetify, plotly, vega, router, markdown, matplotlib, deckgl, simput...) as a dependency (`trame-vtk`, `trame-vuetify`, `trame-*`, ...).

2. The next thing that won't break your project today but will on January 2024 is the change of default client type. You may not know it, but trame is based on vue and support both vue2 and vue3. Right now the default client is using vue2 which has a EOL on December 2023. So we aim to change the default to vue3 in January. Both versions will keep working past that date and core widgets will be made compatible for vue2 and vue3. So the only thing you have to do is to properly assign the client_type that you relying on. So just add the following line to your code base `server.client_type = 'vue2'`.

**Comment by earlsioson** [comment]
@jourdain what happened to the trame module/package from trame.widgets? We are using trame.ClientStateChange and after testing an upgrade to trame v3 we now get this error?

ImportError: cannot import name 'trame' from 'trame.widgets' (/app/.venv/lib/python3.9/site-packages/trame/widgets/__init__.py)

**Reply by jourdain** [reply]
Just need to install `trame-components` by declaring it in your app dependencies.

**Reply by earlsioson** [reply]
Thank you @jourdain, that got me further but now I'm getting this error...

TypeError: Server using client_type='vue3' while we expect 'vue2'

I have set the client type to...

server.client_type = "vue3"

and updated our imports to...

from trame.widgets import trame, vuetify3 as vuetify, paraview

Is there something else I'm missing?

**Reply by jourdain** [reply]
So `trame.widgets.trame` is not vue3 compatible just yet. I am working on it, but still not there fully.

**Reply by earlsioson** [reply]
I see, I'll leave it set for vue2 for now. Thank you.

**Comment by jourdain** [comment]
| Widget name (trame.widget.{name}) | Package name | Supported version |
| --- | --- | --- |
| html | (built-in) | 2 & 3 |
| client | (built-in) | 2 & 3 |
| vuetify, vuetify2, vuetify3 | trame-vuetify | 2 & 3 |
| vtk, paraview | trame-vtk | 2 & 3 |
| plotly | trame-plotly | 2 & 3 |
| xterm | trame-xterm | 2 & 3 |
| code | trame-code | 2 & 3 |
| iframe | trame-iframe | 2 & 3 |
| trame | trame-components | 2 (wip for 3) |
| keycloak | trame-keycloak | 2 & 3 (wip) |
| formkit | trame-formkit | 3 |
| leaflet | trame-leaflet | 2 |
| video | trame-video | 2 |
| pvui | trame-pvui | 2 |
| grid | trame-grid-layout | 2 |
| tauri | trame-tauri | 2 |
| vega | trame-vega | 2 |
| simput | trame-simput | 2 |
| router | trame-router | 2 |
| rca | trame-rca | 2 |
| matplotlib | trame-matplotlib | 2 |
| markdown | trame-markdown | 2 |
| deckgl | trame-deckgl | 2 |
